### ISSUE TITLE
Twitch login windows keeps opening

### ISSUE THREAD

**Issue by flipperyxz** [PedroS11/twitch-live-extension#59]
Please look into extension opening twitch login window. Mine extension keep opening twitch login windows 2-3 minutes when it isn't being authorized. After authorization it is woring fine but in my opinion it should open that window once not infinitely.

**Comment by PedroS11** [comment]
Yeah, every 3minutes the code runs to update the badge icon with the number of live streams but I can have a look at it. I have just one question, is there any reason to have the extension installed but not logged in?

**Comment by flipperyxz** [comment]
Hi thanks for reply, simply answering your question "no" there is no reason
however there is one BUT
your extension is almost perfect, its just sits there and works as it
should be. One problem is that twitch needs authorization every 30 days, i
know im probably too lazy :D but im using password generator and two
authentication method which is just pain in the ass to do whole process.
so i would like to just use browser sometimes whithout login in to
extension which is not really my priority. Alternatively it would be nice
to know when twitch will need another authorization, like 3 days countdown
before need to log in?
Thank you once again and have a nice day!

sob., 18 mar 2023 o 12:44 Pedro Silva ***@***.***> napisał(a):

**Comment by PedroS11** [comment]
Ahh so I think there's a solution for you, when you log in with Twitch, if you don't tick the "Login this computer for 30 days", twitch will keep you logged in forever so you don't need to do it every 30 days. Do you mind trying it, please? I think it should solve your use case and, as soon as I have some free time, I'll have a look at the code to fix that little bug

**Comment by PedroS11** [comment]
I'm struggling to recreate this issue, I have the extension installed, logged out of the account and I got nothing popping. Did you do anything else to have it?

**Comment by flipperyxz** [comment]
Hmm… nothing that I can think of right now. I will check it later on and if it will happen again I’ll let you know Sent from my iPhoneOn 19 Mar 2023, at 18:03, Pedro Silva ***@***.***> wrote:﻿
I'm struggling to recreate this issue, I have the extension installed, logged out of the account and I got nothing popping. Did you do anything else to have it?

**Comment by PedroS11** [comment]
Ok ok, sweet, thank you for it 😄

**Comment by flipperyxz** [comment]
Hi ok I replicated the “problem” although it looks like it opening those windows slower (?). What I did basically I’ve logged in to twitch in my web browser and in twitch settings I’ve deleted twitch extension from “connections” tab, it normally would happen after 30 days because it’s maximum time for twitch to hold authorisation. Sent from my iPhoneOn 19 Mar 2023, at 18:09, Pedro Silva ***@***.***> wrote:﻿
Ok ok, sweet, thank you for it 😄

**Comment by Menteroth** [comment]
I keep my computer on all the time and I also have Twitch tabs open along with it.
So everytime I need to renew the login I might not be at home. Sometimes when I get home from work I can have 400 of those windows asking me to login.

You need to make the plugin check if it already has a window open or something because it's super annoying with all those windows, plus that I have to force quit Chrome and then hopefully restore tabs works when I open the browser again. Luckily every restore so far has worked.

Please fix this issue. Love the plugin otherwise.

**Comment by PedroS11** [comment]
Heey,
Sorry about that, I think I have a fix for it, I will test it during the day and, if everything goes ok, I'll send it to Chrome Store and it takes a few days for them to accept it. I'll post here when the new changes are live.

**Comment by Menteroth** [comment]
Awesome. Thanks for the quick reply.

**Comment by PedroS11** [comment]
Still with some tests but hope for a release during next week!

**Comment by PedroS11** [comment]
@Menteroth just a heads up, the fixed version was sent to Google, in a few days i should be live 🎉

**Comment by PedroS11** [comment]
The fix is now live, chrome will take a few days to update to the newest version but it was published. I'll keep this open for two more days. If it didn't fix your issue, just drop a new message here and I'll have another look!
The updated version of the extension is the 4.2.1
´

**Comment by Menteroth** [comment]
Awesome. I can't check for another 25 days or so. I just logged in before posting my comment here.

**Comment by PedroS11** [comment]
Ok ok, I'll close this. If it happens again, add a new comment here or feel free to open a new issue and I'll jump on it :D

**Comment by Menteroth** [comment]
I'm still getting them. First day today again on one of my PCs. Right when I started my PC and then 3-4 came within a 1 minute of each other. Time in-between seems random as well, if that helps. Took a few more minutes in-between now after I just removed them.

**Comment by PedroS11** [comment]
> I'm still getting them. First day today again on one of my PCs. Right when I started my PC and then 3-4 came within a 1 minute of each other. Time in-between seems random as well, if that helps. Took a few more minutes in-between now after I just removed them.

that's so weird, so can you recreate the steps for me please?
You have it installed but not logged in, after a few time the window appears?

Btw, what operating system do you use? Windows? MacOS? Linux?

**Comment by Menteroth** [comment]
It's weird yeah. And I haven't gotten more since before either. I didn't login and I can still click the app to get live streamers. Which I shouldn't, right? I haven't authorized the app again, yet. 

The PCs are using Windows 8.1 and Windows 10. Both with Chrome.

**Comment by PedroS11** [comment]
Hmmm ok ok, when was the last time you logged out of the accounts? like how many months/years ago?

**Comment by PedroS11** [comment]
Do you mind do the following steps?
- Right click on the extension icon
- Select "Inpect Pop-Up"
- And on the console, run this command

`await chrome.storage.local.remove('tle-token')`

If this solves it, let me know please

**Comment by Menteroth** [comment]
Logged out from Twitch? I've only had to login again like when Twitch forced to change password or when I cleared the browser's cache and that's probably very long ago. It's the same account on both machines btw.

Also, I haven't gotten any pop-ups to login all night. I forgot to turn it off before bed. But yeah, if that has to do with me being able to use it without having to login again last night hehe. 

I only get this when I right-click the icon.
![image](https://github.com/PedroS11/twitch-live-extension/assets/24851042/f566756d-7bc5-4844-a156-20facad2d0fd)

I'm off to work now but can help more later.

**Comment by Menteroth** [comment]
Just wanted to add it's still doing multiple login popups, for the last two months. Not trying to rush you or anything. Just sharing the info.

Also, Happy New Year.

**Comment by PedroS11** [comment]
Hey @Menteroth sorry for the late response, on my personal mac, I have logged out and I don't have that issue.
Are you using windows?

My last recommendation would be, uninstall the extension, re install it, log in with your account and then log out and leave it as it is for some testings please

**Comment by Menteroth** [comment]
Earlier reply, "The PCs are using Windows 8.1 and Windows 10. Both with Chrome."

I removed it and added it because it just stayed chewing on loading. It still opens a billion windows asking me to authorize.
I closed them all, clicked the plugin again so I would get a new window. It worked long ago at least, but no more. It just tries to load and nothing happens. When I do this and it just loads I have to remove it and then add it again for it to ask me.

**Comment by PedroS11** [comment]
@Menteroth do you have any kind of adblocker or something not that common? I have this extension running in multiple computers, MacOs and Windows and this isn't an issue. I haven't released any update in months so all these variations of behaviour I suspect aren't directly related to the extension as I cant replicate it

**Comment by Menteroth** [comment]
It popped up so I could login again, without having to remove/add, so that works again at least. And when I said I encountered previous problem was after your last update. I just responded very late. I'll check if it's still acting up, next time.

I use uBlock Origin.

**Comment by Menteroth** [comment]
Yeah. Still repeated windows opening up for authorization. I understand if this is tiresome. Can just let it go since we don't seem to get anywhere. At least it's not very often.

**Comment by Eshua** [comment]
You keep saying you can't replicate this yet it keeps happening to multiple people. I have been asked if running with other apps, specific apps, fresh install, reinstall, updated.

Its clearly not happening to everyone and yet you keep getting people with the same issue. 

Cause guess what, last year was a new computer, chrome with no other apps but this one AND it was freshly added. It. Still. Happened. The fact you still won't change the behaviour of pop-up every minute is absolute insane and malicious at this point. Good for you that you can't get it to happen, doesn't change the fact you're leaving the rest of us with no solution

**Comment by PedroS11** [comment]
https://github.com/PedroS11/twitch-live-extension/pull/76 raised a MR with a new approach to handle authentication errors. Basically, a message with "Auth" will be displayed on the extension icon with red background and a button on the Settins to enable manual token refresh. Would this be better than the current approach? Open to feedback/suggstions

**Comment by PedroS11** [comment]
cc @Menteroth @Eshua
